docs/guides/building-forms.md

```diff
@@ -0,0 +1,387 @@
+
+```js
+export default function PageWithJSbasedForm() {
+  // Handles the submit event on form submit.
+  const handleSubmit = async (event) => {
+    // Stop the form from submitting and refreshing the page.
+    event.preventDefault()
+
+    // Get data from the form.
+    const data = {
+      first: event.target.first.value,
+      last: event.target.last.value,
+    }
+
+    // Send the data to the server in JSON format.
+    const JSONdata = JSON.stringify(data)
+
+    // API endpoint where we send form data.
+    const endpoint = '/api/form'
+
+    // Form the request for sending data to the server.
+    const options = {
+      // The method is POST because we are sending data.
+      method: 'POST',
+      // Tell the server we're sending JSON.
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      // Body of the request is the JSON data we created above.
+      body: JSONdata,
+    }
+
+    // Send the form data to our forms API on Vercel and get a response.
+    const response = await fetch(endpoint, options)
+
+    // Get the response data from server as JSON.
+    // If server returns the name submitted, that means the form works.
+    const result = await response.json()
+    alert(`Is this your full name: ${result.data}`)
+  }
+  return (
+    // We pass the event to the handleSubmit() function on submit.
+    <form onSubmit={handleSubmit}>
+      <label htmlFor="first">First Name</label>
+      <input type="text" id="first" name="first" required />
+
+      <label htmlFor="last">Last Name</label>
+      <input type="text" id="last" name="last" required />
+
+      <button type="submit">Submit</button>
+    </form>
+  )
+}
+```
+
+It's a Next.js page with a React function component called `PageWithJSbasedForm` with a `<form>` element written in JSX. There's no action on the `<form>` element. Instead, we use the `onSubmit` event handler to send data to our `{handleSubmit}` function.
+
+The `handleSubmit()` function processes your form data through a series of steps:
+
+- The `event.preventDefault()` stops the `<form>` element from refreshing the entire page.
+- We created a JavaScript object called `data` with the `first` and `last` values from the form.
+- JSON is a language-agnostic data transfer format. So we use `JSON.stringify(data)` to convert the data to JSON.
+- We then use `fetch()` to send the data to our `/api/form` endpoint using JSON and HTTP `POST` method.
+- Server sends back a response with the name submitted. Woohoo! 🥳
+
+<video autoplay loop muted playsinline>
+  <source src="https://assets.vercel.com/video/upload/v1643009088/nextjs/guides/building-forms/form-with-js.mp4" type="video/mp4">
+</video>
+
+## Conclusion
+
+Next.js and Vercel provide you with an all-in solution for hosting web forms and handling submissions with serverless functions. Use Next.js as the client for your JavaScript-based interaction and Vercel for its powerful serverless functions to handle the server.
```

Comment:
I wonder if the conclusion should first talk about the fact that we have learnt how to build forms in html, then react, then serverless etc... The conclusion takes in the whole guide, and only a small section was to do with vercel ☺️

Reply:
Yes, I believe that too. We should give a more concrete ending. 💯